src/Tools/RepoUtil/README.md

```diff
@@ -2,24 +2,29 @@
 
 ## Usage
 
-This is a tool used to manage the inputs and outputs of our repo.  In particular, it is used to manage our use of NuGet packages.  It helps to automate the otherwise tedious process of updating NuGet references by:
+This is a tool that manages the use of NuGet packages in the repo:
 
-- Ensuring consistency in NuGet package references
-- Fixing up project.json files
+- Ensuring we consistently use the same package version in our projects.  
+- Ensuring we aren't referencing stale / deleted package versions in configuration.
 - Generating helper files with specified package versions. Example [Dependencies.props](https://github.com/dotnet/roslyn/blob/master/build/Targets/Dependencies.props)
+- Simple way to upgrade package versions in the repo.
 
-The tool works by using all of our project.json files as the primary source of truth for the repo.  All files with the name pattern `*project.json` are considered to be NuGet assets and will be scanned for NuGet references.  
+The primary goal of the tool is to ensure consistency in our NuGet package usage.  Given the large number of project.json files in our repo and the variety of ship vehicles it's easy for packages to get out of sync.  For instance referencing two versions of System.Collection.Immutable.  Doing so is is incorrect, potentially invalidates our testing and potentially breaks insertions. 
 
-This will impose a minimum of requirements on our NuGet packages.  In particular the tool assumes that all references to a given NuGet package should occur at the same version.  If the tool sees a package being used with different versions it will issue an error.  
-
-There are valid cases where a package is used at more than one version in the repo.  Typically because it's being deployed as an asset, used as a tool, etc ...  In those cases an entry can be added to [RepoData.json](https://github.com/dotnet/roslyn/blob/master/src/Tools/RepoUtil/RepoData.json)) in the fixed table to call out that usage:
+The tool ensures that by default given package is only referenced at a single version for the entire repo.  That is we can only use a single version of System.Collections.Immutable in our shipping code.  Deviatons are allowed but they must be explicitly added to the [config file](https://github.com/dotnet/roslyn/blob/master/build/config/RepoUtilData.json) as a "fixed" package.  There is nothing inherently wrong with adding a fixed package so long as it's not used in shipping code.  It's perfectly fine for tools, assets, etc ... 
```

Comment:
I believe your intended statement is correct but I suspect as phrased this misses some subtlety. For the IDE, it's entirely acceptable and intended for us to have different packages referencing different versions; consider our Dev14 vs. Dev15 split. Put another way: right now our repo violates this statement you're making.